posts/2022-05-03-microstream-cdi-ext.adoc

```diff
@@ -0,0 +1,271 @@
+- name: Otavio Santana
+  github: https://github.com/otaviojava
+  image: https://avatars0.githubusercontent.com/otaviojava
+---
+= Easy and Fast Data Access in Open Liberty with MicroStream CDI Extension
+Rudy De Busscher <https://github.com/rdebusscher>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+// // // // // // // //
+// In the preceding section:
+// Do not insert any blank lines between any of the lines.
+//
+// "open-graph-image" is set to OL logo. Whenever possible update this to a more appriopriate/specific image (for example if present an image that is being used in the post). 
+// However, it can be left empty which will set it to the default
+//
+//
+// Replace DESCRIPTION with a short summary (~60 words) of the release (a more succinct version of the first paragraph of the post).
+//
+//
+// If adding image into the post add :
+// -------------------------
+// [.img_border_light]
+// image::img/blog/FILE_NAME[IMAGE CAPTION ,width=70%,align="center"]
+// -------------------------
+// "[.img_border_light]" = This adds a faint grey border around the image to make its edges sharper. Use it around
+// screenshots but not around diagrams. Then double check how it looks.
+// There is also a "[.img_border_dark]" class which tends to work best with screenshots that are taken on dark backgrounds.
+// Once again make sure to double check how it looks
+// Change "FILE_NAME" to the name of the image file. Also make sure to put the image into the right folder which is: img/blog
+// change the "IMAGE CAPTION" to a couple words of what the image is
+// // // // // // // //
+
+Microservices has become a buzzword when we talk about creating a scalable application. As with any software architecture decision, it has trade-offs and challenges. One of those challenges is the distributed nature of the environment.  Data access is another challenge. Since many solutions make use of data that is stored in an entirely different format as the Java Objects, such as a database with tables and columns, accessing data can be slow due to the mapping that is required.
+
+Open Liberty is a great runtime for hosting microservices. It interacts well with many data access libraries. In this blog, we will look at how Open Liberty can work well with the https://microstream.one/platforms/microstream-for-java/[MicroStream framework] to demonstrate the fast data access within microservices.
+Let’s first have a quick tour of MicroStream.  
+
+== Introduction of MicroStream
+
+MicroStream allows you to write a Java Object graph very efficiently and fast to data storage. This can be the disk, blob storage of a database, or any other medium. Since your data is stored in memory, it provides microsecond query time, low-latency data access, and gigantic data throughput since there is no mapping as the data are just Java instances. There is no mapping and conversion to a database system needed.
```

Comment:
```suggestion
MicroStream allows you to write a Java Object graph to data storage very efficiently and quickly. This storage can be the disk, blob storage of a database, or any other medium. Since your data is stored in memory, it provides microsecond query time, low-latency data access, and gigantic data throughput because the data are just Java instances. No mapping and conversion to a database system is needed.
```